students/avalvillar/lesson7/html_render.py

```diff
@@ -0,0 +1,143 @@
+"""
+    Author is Antonio V. Alvillar
+    Self-Paced-Online Python 210 UWPCE
+    January 16th 2019
+"""
+
+'''
+This is the Element class that is subclassed from 'object' and will have many
+subclasses from itselt. The overall goal is to be able to render HTML
+documentation as functions are called for particular tags, attributes, and
+indentations/levels.
+'''
+
+#Top level class of project.
+class Element(object):
+
+    #Class level attributes for an empty tag, default indentations
+    tag = ''
+    indentation = 4
+    attrs = {'':''}
+
+    #If attributes are given - put them in self.attrs and if contens is None
+    #create an empty content list to append to later.
+    def __init__(self, content=None, **kwargs):
+        self.attrs = kwargs
+        if content is None:
+            self.contents = []
+        else:
+            self.contents = [content]
+
+    #Append function to add contents as needed to a list for later use.
+    def append(self, content):
+        self.contents.append(content)
+
+    #Render method that will write to the file object passed in.
+    def render(self, file_out, cur_ind=''):
+        '''if cur_ind != '':
+            file_out.write(' ' * cur_ind)'''
+        if self.attrs.keys():
+            file_out.write('<' + self.tag)
+            for k, v in self.attrs.items():
+                file_out.write(' {}=\"{}\"'.format(k,v))
+            file_out.write('>')
+        else:
+            file_out.write('<' + self.tag + '>\n')
+        for item in self.contents:
+            try:
+                item.render(file_out, self.indentation * 2)
+            except AttributeError:
+                file_out.write(item)
+        '''if cur_ind != '':
+            file_out.write(' ' * cur_ind)'''
+        file_out.write('\n</' + self.tag + '>')
+
+#Subclass of Element - will add "Doctype" to all new HTML files.
+class Html(Element):
+    tag = 'html'
+    def render(self, file_out, cur_ind=""):
+        file_out.write('<!DOCTYPE html>\n')
+        Element.render(self, file_out)
+
+#Subclass of Element
+class Body(Element):
+    tag = 'body'
+    def render(self, file_out, cur_ind=""):
+        Element.render(self, file_out)
```

Comment:
you don't need define or call parent render() method - you're not adding anything different in your `Body` subclass.